camel/models/reka_model.py

```diff
@@ -0,0 +1,212 @@
+        self,
+        model_type: ModelType,
+        model_config_dict: Dict[str, Any],
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+        token_counter: Optional[BaseTokenCounter] = None,
+    ) -> None:
+        r"""Constructor for Reka backend.
+
+        Args:
+            model_type (ModelType): Model for which a backend is created,
+                one of REKA_* series.
+            model_config_dict (Dict[str, Any]): A dictionary that will
+                be fed into `Reka.chat.create`.
+            api_key (Optional[str]): The API key for authenticating with the
+                Reka service. (default: :obj:`None`)
+            url (Optional[str]): The url to the Reka service.
+            token_counter (Optional[BaseTokenCounter]): Token counter to use
+                for the model. If not provided, `OpenAITokenCounter` will be
+                used.
+        """
+        super().__init__(
+            model_type, model_config_dict, api_key, url, token_counter
+        )
+        self._api_key = api_key or os.environ.get("REKA_API_KEY")
+        self._url = url or os.environ.get("REKA_SERVER_URL")
+
+        from reka.client import Reka
+
+        self._client = Reka(api_key=self._api_key, base_url=self._url)
+        self._token_counter: Optional[BaseTokenCounter] = None
+
+    def _to_openai_response(self, response: 'ChatResponse') -> ChatCompletion:
+        obj = ChatCompletion.construct(
+            id=response.id,
+            choices=[
+                dict(
+                    message={
+                        "role": response.responses[0].message.role,  # type: ignore[index,union-attr]
+                        "content": response.responses[0].message.content,  # type: ignore[index,union-attr]
+                    },
+                    finish_reason=response.responses[0].finish_reason  # type: ignore[index]
+                    if response.responses[0].finish_reason  # type: ignore[index]
+                    else None,
+                )
+            ],
+            created=None,
+            model=response.model,
+            object="chat.completion",
+            usage=response.usage,
+        )
+
+        return obj
+
+    def _to_reka_chatmessage(
+        self,
+        messages: List[OpenAIMessage],
+    ) -> List["ChatMessage"]:
+        from reka.types import ChatMessage
+
+        new_messages = []
+        for msg in messages:
+            role = msg.get("role")
+            content = msg.get("content")
+
+            if role == "user":
+                new_messages.append(ChatMessage(role="user", content=content))  # type: ignore[arg-type]
+            elif role == "assistant":
+                new_messages.append(
+                    ChatMessage(role="assistant", content=content)  # type: ignore[arg-type]
+                )
+            elif role == "system":
+                new_messages.append(ChatMessage(role="user", content=content))  # type: ignore[arg-type]
```

Comment:
I noticed there are several ‘type: ignore’ statements throughout the code. Do we have a better way to address these issues?

Reply:
Seems they updated the SDK, currently there's no need to add that much ignore for type checking